### PR TITLE
#10333 Template Updates

### PR DESCRIPTION
**Pull request by isaacrlevin**
Summary of the changes (Less than 80 chars)
 - Updates to C# and F# WebApi project to have WeatherController instead of ValuesController

Addresses #10333

### CODE REVIEW

File: src/ProjectTemplates/Web.ProjectTemplates/content/WebApi-CSharp/Controllers/WeatherController.cs
Comment (ryanbrandenburg) [inline src/ProjectTemplates/Web.ProjectTemplates/content/WebApi-CSharp/Controllers/WeatherController.cs]: Duplicate member name with above.

File: src/ProjectTemplates/Web.ProjectTemplates/content/WebApi-FSharp/Controllers/WeatherController.fs
Comment (ryanbrandenburg) [inline src/ProjectTemplates/Web.ProjectTemplates/content/WebApi-FSharp/Controllers/WeatherController.fs]: This template explicitly compiles each of its items, so [the fsproj.in](https://github.com/aspnet/AspNetCore/blob/master/src/ProjectTemplates/Web.ProjectTemplates/WebApi-FSharp.fsproj.in) is still pointed at the now-missing ValuesController.fs.

File: src/ProjectTemplates/Web.ProjectTemplates/content/WebApi-CSharp/Controllers/ValuesController.cs
Comment (ryanbrandenburg) [inline src/ProjectTemplates/Web.ProjectTemplates/content/WebApi-CSharp/Controllers/ValuesController.cs]: You'll need to [update the baselines](https://github.com/aspnet/AspNetCore/blob/master/src/ProjectTemplates/test/template-baselines.json#L398) to reflect the new file names.

**Review by ryanbrandenburg — CHANGES_REQUESTED** [review]
Changes are good in principle, but cause [some test failures](https://dev.azure.com/dnceng/public/_build/results?buildId=194241&view=ms.vss-test-web.build-test-results-tab) that you'll need to react to.

**Comment by ryanbrandenburg** [comment]
More failures, looks like compile failures in the fsharp project and we need to change the expected URL in the WebApi tests.

**Comment by isaac2004** [comment]
@ryanbrandenburg not sure what is going on here, but the build errors are related to code I have not touched

```

EXEC : warning : An illegal reflective access operation has occurred [F:\workspace\_work\1\s\src\SignalR\clients\java\signalr\signalr.client.java.javaproj]
EXEC : warning : Illegal reflective access by org.codehaus.groovy.vmplugin.v7.Java7$1 (file:/C:/Users/vsagent/.gradle/wrapper/dists/gradle-5.3-bin/9tevgwefmd5lgaxkzkt5jgvch/gradle-5.3/lib/groovy-all-1.0-2.5.4.jar) to constructor java.lang.invoke.MethodHandles$Lookup(java.lang.Class,int) [F:\workspace\_work\1\s\src\SignalR\clients\java\signalr\signalr.client.java.javaproj]
EXEC : warning : Please consider reporting this to the maintainers of org.codehaus.groovy.vmplugin.v7.Java7$1 [F:\workspace\_work\1\s\src\SignalR\clients\java\signalr\signalr.client.java.javaproj]
EXEC : warning : Use --illegal-access=warn to enable warnings of further illegal reflective access operations [F:\workspace\_work\1\s\src\SignalR\clients\java\signalr\signalr.client.java.javaproj]
EXEC : warning : All illegal access operations will be denied in a future release [F:\workspace\_work\1\s\src\SignalR\clients\java\signalr\signalr.client.java.javaproj]
F:\workspace\_work\1\s\src\Security\Authentication\Negotiate\samples\NegotiateAuthSample\Startup.cs(18,22): error CS1061: 'IServiceCollection' does not contain a definition for 'AddAuthorization' and no accessible extension method 'AddAuthorization' accepting a first argument of type 'IServiceCollection' could be found (are you missing a using directive or an assembly reference?) [F:\workspace\_work\1\s\src\Security\Authentication\Negotiate\samples\NegotiateAuthSample\NegotiateAuthSample.csproj]
F:\workspace\_work\1\s\src\Security\Authentication\Negotiate\samples\NegotiateAuthSample\Startup.cs(30,17): error CS1061: 'IApplicationBuilder' does not contain a definition for 'UseAuthorization' and no accessible extension method 'UseAuthorization' accepting a first argument of type 'IApplicationBuilder' could be found (are you missing a using directive or an assembly reference?) [F:\workspace\_work\1\s\src\Security\Authentication\Negotiate\samples\NegotiateAuthSample\NegotiateAuthSample.csproj]
    5 Warning(s)
    2 Error(s)
```

**Comment by ryanbrandenburg** [comment]
Fetch latest and rebase, [conflicting PR's resulted in compiler errors](https://github.com/aspnet/AspNetCore-Internal/issues/2554#issuecomment-494865185). Should be fixed on the latest commits.

File: src/ProjectTemplates/Web.ProjectTemplates/content/WebApi-CSharp/Controllers/WeatherController.cs
Comment (ryanbrandenburg) [inline src/ProjectTemplates/Web.ProjectTemplates/content/WebApi-CSharp/Controllers/WeatherController.cs]: I was about to say that generally you [want to re-use](https://docs.microsoft.com/en-us/dotnet/api/system.random?redirectedfrom=MSDN&view=netcore-3.0#avoiding-multiple-instantiations) the same `Random` instance rather than re-creating it on each call, but it turns out that `Random` [is not thread-safe](https://docs.microsoft.com/en-us/dotnet/api/system.random?redirectedfrom=MSDN&view=netcore-3.0#the-systemrandom-class-and-thread-safety), meaning that you wouldn't want to use just the one instance without a doing your own locking scheme.

All of that's just background for anyone interested though, there's no need to change what you have here since it's just a sample and doesn't need to be production ready.

**Review by ryanbrandenburg — APPROVED** [review]
Looks great! I'll merge this once the Helix build finishes.

**Comment by isaac2004** [comment]
@ryanbrandenburg build finally passes. Let me know if you have any questions.

**Comment by ryanbrandenburg** [comment]
@aspnet-hello triage https://dev.azure.com/dnceng/public/_build/results?buildId=207058&view=results

**Comment by aspnet-hello** [comment]
This comment was made automatically. If there is a problem contact aspnetcore-build@microsoft.com.

I've triaged the above build. I've created/commented on the following issue(s)
https://github.com/aspnet/AspNetCore-Internal/issues/1989
https://github.com/aspnet/AspNetCore-Internal/issues/2570

**Comment by isaac2004** [comment]
@ryanbrandenburg is this blocking the merge?

**Pull request merged**
